### PR TITLE
GEP 10 - ContainerD Enablement

### PR DESCRIPTION
**Pull request by nimrodoron**
Enablement of ContainerD according to GEP-10

**Special notes for your reviewer**:
Depends on: https://github.com/gardener/gardener-extensions/pull/617 + revendor

### CODE REVIEW

File: pkg/coreos/actuator_reconcile.go [hunk 1]
```diff
@@ -97,6 +97,31 @@ func (c *actuator) cloudConfigFromOperatingSystemConfig(ctx context.Context, con
 			})
 		}
 
+		if unit.Name == "containerd.service" && config.Spec.EnableContainerD {
```
Comment (rfranzke) [inline pkg/coreos/actuator_reconcile.go]: Can we add a constant for this service name like for the kubelet: https://github.com/gardener/gardener/blob/master/pkg/apis/core/v1beta1/constants/types_constants.go#L238-L239 and use it?

File: pkg/coreos/templates/containerd-exec-start [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[Service]
+ExecStart=
```
Comment (rfranzke) [inline pkg/coreos/templates/containerd-exec-start]: What's the reason for this?

File: pkg/coreos/actuator.go [hunk 3]
```diff
@@ -29,11 +30,14 @@ type actuator struct {
...
 	logger logr.Logger
+	box    *packr.Box
 }
 
+//go:generate packr2
+
 // NewActuator creates a new Actuator that updates the status of the handled OperatingSystemConfigs.
 func NewActuator() operatingsystemconfig.Actuator {
-	return &actuator{logger: log.Log.WithName("coreos-operatingsystemconfig-actuator")}
+	return &actuator{logger: log.Log.WithName("coreos-operatingsystemconfig-actuator"), box: packr.New("templates", "./templates")}
```
Comment (rfranzke) [inline pkg/coreos/actuator.go]: Why do we have to store these data in separate files? Can't we simply inject the content here? Then we don't need packr/box etc. WDYT?

File: pkg/coreos/actuator_reconcile.go [hunk 4]
```diff
@@ -97,6 +97,31 @@ func (c *actuator) cloudConfigFromOperatingSystemConfig(ctx context.Context, con
...
+				Name:    "11-exec-start-config.conf",
+				Content: string(containerdUnit),
+			})
+		}
+
+		cloudConfig.CoreOS.Units = append(cloudConfig.CoreOS.Units, u)
+	}
+
+	if config.Spec.Purpose == extensionsv1alpha1.OperatingSystemConfigPurposeReconcile && config.Spec.EnableContainerD {
+		u := Unit{Name: "docker.service"}
```
Comment (rfranzke) [inline pkg/coreos/actuator_reconcile.go]: Same here, we probably should add a constant.

File: pkg/coreos/templates/containerd-exec-start (hunk 2 above)
Comment (nimrodoron) [reply pkg/coreos/templates/containerd-exec-start]: To override the ExecStart it must be clear first.

File: pkg/coreos/actuator_reconcile.go (hunk 1 above)
Comment (nimrodoron) [reply pkg/coreos/actuator_reconcile.go]: Done

File: pkg/coreos/actuator.go (hunk 3 above)
Comment (nimrodoron) [reply pkg/coreos/actuator.go]: Done

File: pkg/coreos/actuator_reconcile.go (hunk 4 above)
Comment (nimrodoron) [reply pkg/coreos/actuator_reconcile.go]: Done

**Comment by nimrodoron** [comment]
This PR depends on gardener/gardener#1971 and gardener/gardener-extensions#604 + revendoring

File: pkg/coreos/const.go [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package coreos
```
Comment (rfranzke) [inline pkg/coreos/const.go]: license header missing

File: pkg/coreos/actuator_reconcile.go [hunk 6]
```diff
@@ -74,6 +75,8 @@ func (c *actuator) cloudConfigFromOperatingSystemConfig(ctx context.Context, con
 		}
 	}
 
+	enableContainerD := config.Spec.CRIConfig != nil && config.Spec.CRIConfig.Name == extensionsv1alpha1.CRINameContainerD
```
Comment (danielfoehrKn) [inline pkg/coreos/actuator_reconcile.go]: Field does not exist in config. Did you maybe forget to commit a file?

Comment (danielfoehrKn) [reply pkg/coreos/actuator_reconcile.go]: ok missed that it needs revendor

**Pull request closed without merging**